src/Infrastructure/Controller/GetMapFilterController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Infrastructure\Controller;
+
+use App\Domain\Regulation\Repository\LocationRepositoryInterface;
+use App\Infrastructure\Form\Map\MapFilterFormType;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapQueryParameter;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Routing\RouterInterface;
+
+final class GetMapFilterController extends AbstractController
+{
+    public function __construct(
+        private \Twig\Environment $twig,
+        private FormFactoryInterface $formFactory,
+        private RouterInterface $router,
+        private LocationRepositoryInterface $locationRepository,
+    ) {
+    }
+
+    #[Route(
+        '/_get_map_filter',
+        name: 'get_map_filter',
+        methods: ['GET'],
+    )]
```

Comment:
Je crois comprendre que ce controller renvoie d'une part le nouvel `<output>`, et d'autre part le nouvel état du formulaire

Dès lors qu'on a accès à JavaScript, je pense qu'on peut se satisfaire de seulement renvoyer le textContent du `<output>`, car le formulaire va se changer visuellement de lui-même quand on le manipule

Autre remarque, ne pas utiliser "get" dans l'URL (l'action "get" est déjà sémantiquement incluse dans le fait qu'on utilise la méthode HTTP "GET"), et coller mieux au naming usuel des URLs et routes qui renvoient un "fragment" (pas une page complète)

Concernant le parsing des query params, j'avoue que pour l'instant on le faisait un peu "à la main" en inspectant `$request->query`. La syntaxe avec l'annotation `#[MapQueryParameter]` me paraît un peu obscure

Ce qui donnerait donc plutôt ça...

```php
    #[Route(
        '/_fragments/map/locations',
        name: 'fragment_map_locations',
        methods: ['GET'],
    )]
public function __invoke(Request $request): Response
    {
        $category = $request->query->get('category', 'permanents_and_temporaries');
        $upcomingOnly = $request->query->get('upcomingOnly') === 'true';

        $geoJson = $this->locationRepository->findAllForMapAsGeoJSON(
            includePermanent: in_array($category, ['permanents_and_temporaries', 'permanents'],
            includeTemporary: in_array($category, ['permanents_and_temporaries', 'temporaries'],
           upcomingOnly: $upcomingOnly,
        );

        return new Response(
            $this->twig->render(
                name: 'map/fragment/locations.html.twig',
                context: [
                    'geojson' => $geoJson,
                ],
            ),
        );
    }
```

Et `map/fragment/locations.html.twig` contiendrait un Turbo Stream pour remplacer le contenu de l'`<output>` (l'action "update" permet de remplacer le contenu d'un élément, sans remplacer l'élément lui-même)

```html
<turbo-stream target="map-locations-data" action="update">
    <template>{{ geoJson }}</template>
</turbo-stream>

Reply:
> Concernant le parsing des query params, j'avoue que pour l'instant on le faisait un peu "à la main" en inspectant $request->query. La syntaxe avec l'annotation #[MapQueryParameter] me paraît un peu obscure

Je serais d'accord pour utiliser ces annotations : ca devient la bonne pratique côté Symfony https://symfony.com/blog/new-in-symfony-6-3-query-parameters-mapper